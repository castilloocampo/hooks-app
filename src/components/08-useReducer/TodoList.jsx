import React from 'react';
import { PropTypes } from 'prop-types';
import TodoListItem from './TodoListItem';

const TodoList = ({ todos, onHandleToggleTODO, onHandleDeleteTODO }) => {
  return (
    <ul>
      {
        todos.map((todo, i) =>
          <TodoListItem 
            key={todo.id}
            todo={todo} 
            index={i} 
            onHandleDeleteTODO={onHandleDeleteTODO}
            onHandleToggleTODO={onHandleToggleTODO} />
        )
      }
    </ul>
  );
}

TodoList.propTypes = {
  todos: PropTypes.array.isRequired,
  onHandleToggleTODO: PropTypes.func.isRequired,
  onHandleDeleteTODO: PropTypes.func.isRequired,
}

export default TodoList;
