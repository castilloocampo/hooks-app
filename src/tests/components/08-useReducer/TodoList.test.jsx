import { render, screen } from '@testing-library/react';
import TodoList from '../../../components/08-useReducer/TodoList';
import { demoTodos } from '../../fixtures/demoTodos';


describe('TodoList', () => {
  const toggleTodo = jest.fn();
  const deleteTodo = jest.fn();
  let todoRendered = render(<TodoList
    todos={demoTodos}
    onHandleToggleTODO={toggleTodo}
    onHandleDeleteTODO={deleteTodo}
  />);

  let { container } = todoRendered;

  beforeEach(() => {
    todoRendered = render(<TodoList
      todos={demoTodos}
      onHandleToggleTODO={toggleTodo}
      onHandleDeleteTODO={deleteTodo}
    />);
    container = todoRendered.container;
  });

  test('should show correctly', () => {
    expect(container).toMatchSnapshot();
  });

  test('should have two elements of todoListItem', () => {
    const prgph = screen.getAllByTestId('p-list-item');
    expect(prgph.length).toBe(2);
  })
  
})
