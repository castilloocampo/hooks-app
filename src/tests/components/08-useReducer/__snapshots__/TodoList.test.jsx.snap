// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList should show correctly 1`] = `
<div>
  <ul>
    <li
      class="relative block py-3 px-6 -mb-px "
      id="list-group-item"
    >
      <p
        class="text-center false"
        data-testid="p-list-item"
      >
        1
        . 
        Learn React
      </p>
      <button
        class="btn btn-danger"
        data-testid="delete-button"
      >
        Borrar
      </button>
    </li>
    <li
      class="relative block py-3 px-6 -mb-px "
      id="list-group-item"
    >
      <p
        class="text-center false"
        data-testid="p-list-item"
      >
        2
        . 
        Learn Mongo
      </p>
      <button
        class="btn btn-danger"
        data-testid="delete-button"
      >
        Borrar
      </button>
    </li>
  </ul>
</div>
`;
